File: CivitasJava/civitas/Casilla.java
package civitas;

import java.util.ArrayList;

public class Casilla {

    private String nombre;

    Casilla(String nombre) 
    {
        this.nombre = nombre;
    }

    public String getNombre() 
    {
        return this.nombre;
    }
    protected void informe(int iactual, ArrayList<Jugador> todos) 
    {
        Diario.getInstance().ocurreEvento("El jugador " + todos.get(iactual).getNombre() + " ha caido en la casilla " + this.toString() + "\n");
    }

    public boolean jugadorCorrecto(int iactual, ArrayList<Jugador> todos) 
    {
        return iactual >= 0 && iactual < todos.size();
    }

    void recibeJugador(int iactual, ArrayList<Jugador> todos)
    {
        if(this.jugadorCorrecto(iactual, todos))
        {
            this.informe(iactual, todos);
        }
    }

    public String toString() 
    {
        String ret = this.nombre + " de tipo descanso.\n";
        return ret;
    }
}

File: CivitasRuby/Casilla.rb
module Civitas

    class Casilla
        attr_reader :nombre
        
        def initialize(nombre)
            @nombre=nombre
        end

        def informe(actual, todos)
        
            Diario.instance.ocurre_evento('El jugador ' + todos[actual].nombre + ' ha caido en la casilla ' + toString)    

        end

        def jugadorCorrecto(actual, todos)
        
            return actual>=0 && actual<todos.length
        
        end
        
        def recibeJugador(actual, todos)

            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
            end 
        end
        
        def toString
            ret=@nombre + ' de tipo descanso'
            return ret
        end

        protected :informe 

    end
end